sdk/core/core-lro/src/poller/poller.ts

```diff
@@ -201,7 +220,24 @@ export function buildCreatePoller<TResponse, TResult, TState extends OperationSt
               throw state.error;
           }
         }
+
+        return state;
+      },
+      then<TResult1 = TResult, TResult2 = never>(
+        onfulfilled?: ((value: TResult) => TResult1 | PromiseLike<TResult1>) | undefined | null,
+        onrejected?: ((reason: any) => TResult2 | PromiseLike<TResult2>) | undefined | null,
+      ): Promise<TResult1 | TResult2> {
+        return poller.pollUntilDone().then(onfulfilled, onrejected);
+      },
+      catch<TResult2 = never>(
+        onrejected?: ((reason: any) => TResult2 | PromiseLike<TResult2>) | undefined | null,
+      ): Promise<TResult | TResult2> {
+        return poller.pollUntilDone().catch(onrejected);
```

Comment:
Can you help me double check that chaining `catch` and `finally` don't cause the `poller.pollUntilDone` to be executed twice?

Reply:
they would cause the poller.pollUntilDone twice but the return value of pollUntilDone actually is the same promise and you could check the logic https://github.com/Azure/azure-sdk-for-js/blob/ee44d20988c4caf14f39f82a5837d20114ccc090/sdk/core/core-lro/src/poller/poller.ts#L136.